### DISCUSSION TITLE
Is there a way to do dynamic redirects and errors?

### DISCUSSION THREAD

**Discussion by cipami** [nuxt/nuxt#17226] [Questions]
in Nuxt2 there were `context.errror()` and `context.redirect()` functions. I need to use them dynamically from the `useFetch()` response. What is the best practice to do this in Nuxt3?

**Comment by elHornair** [comment]
What is suggested [here](https://github.com/nuxt/framework/discussions/906) (for a slightly different question though), is to use a [Server Middleware](https://v3.nuxtjs.org/docs/directory-structure/server/#server-middleware).
This should work fine, if no additional info from the nuxt context is needed.

**Comment by psycongaroo** [comment]
@danielroe 
is possible do it without Server Middleware?

like:
```
<script setup>
const nuxtApp = useNuxtApp();
const { data, error } = useFetch('/test');

if(error.value)}{
   nuxtApp.error({statusCode:404});
}
</script>
```

**Reply by wobsoriano** [reply]
https://github.com/nuxt/framework/discussions/2541#discussioncomment-1887727

**Reply by psycongaroo** [reply]
@wobsoriano 
Tried and is not working:

```
`<script setup>
const { data, error } = useFetch('/test');

if(error.value){
   const error = new Error(`Page not found: ${useRoute().fullPath}`)
   error.statusCode = 404
   throw error
}
</script>`
```

**Reply by Bjornnyborg** [reply]
@shinguweb96 Maybe this could give you a hint: I am using this in a nuxt3 project, where i fetch content from an API with the: `useAsyncData()`
If the API returns `404`, i fetch content for the "not found page", and returns a `404` status.
```js
const { ssrContext } = useNuxtApp();

if (process.server) {
   const error = new Error();
   (error as any).statusCode = 404;
   (ssrContext as any).nuxt.error = error;
}
```
[See file here](https://github.com/enterspeedhq/enterspeed-demo-nuxt3/blob/4c97b6de6845dc143df036d40b35e67c0b38a4fc/enterspeed-connect.ts#L90)

**Comment by Lanayx** [comment]
It seems there is a `navigateTo` function available globally ([docs](https://v3.nuxtjs.org/docs/directory-structure/middleware))

**Comment by mtskf** [comment]
How about this...?
```
<script setup>
import {createError} from "h3"
const { data: count } = await useFetch('/api/count').catch((e) => {
  throw createError({ statusCode: e.response.status, statusMessage:  e.response.statusText })
})
</script>
```
